src/main/java/com/google/sps/utility/GmailUtility.java

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utility;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.model.Message;
+import com.google.sps.model.MessageFormat;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Contains logic that handles GET & POST requests to the Gmail API and transforms those responses
+ * into easily usable Java types
+ */
+public class GmailUtility {
+  // Make constructor private so no instances of this class can be made
+  private GmailUtility() {}
+
+  /**
+   * Creates a query string for Gmail. Use in search to return emails that fit certain restrictions
+   *
+   * @param emailAge emails from the last emailAge [emailAgeUnits] will be returned. Set to 0 to
+   *     ignore filter
+   * @param emailAgeUnits "d" for days, "h" for hours, "" for ignore email
+   * @param unreadOnly true if only returning unread emails, false otherwise
+   * @param from email address of the sender. "" if not specified
+   * @return string to use in gmail (either client or API) to find emails that match criteria
+   */
+  public static String emailQueryString(
+      int emailAge, String emailAgeUnits, boolean unreadOnly, String from) {
+    String queryString = "";
+
+    // Add query components
+    queryString += emailAgeQuery(emailAge, emailAgeUnits);
+    queryString += unreadEmailQuery(unreadOnly);
+    queryString += fromEmailQuery(from);
+
+    // Return multi-part query
+    return queryString;
+  }
+
+  /**
+   * Creates Gmail query for age of emails
+   *
+   * @param emailAge emails from the last emailAge [emailAgeUnits] will be returned. Set to 0 to
+   *     ignore filter
+   * @param emailAgeUnits "d" for days, "h" for hours, "" for ignore email
+   * @return string to use in Gmail (either client or API) to find emails that match these criteria
+   *     or null if either one of the arguments are invalid Trailing space added to properties so
+   *     multiple queries can be concatenated
+   */
+  public static String emailAgeQuery(int emailAge, String emailAgeUnits) {
+    // newer_than:#d where # is an integer will specify to only return emails from last # days
+    if (emailAge > 0 && (emailAgeUnits.equals("h") || emailAgeUnits.equals("d"))) {
```

Comment:
I see [here](https://support.google.com/mail/answer/7190?hl=en) they define the units as d, m, and y. Are you sure h works?

Reply:
Strange that it isn't defined there - it does work in Gmail's search bar, which means it would also work for the API (the query selector is the same as using the search bar).

I did however (on another note) realize I've added spaces to the search query where it is not permitted. Fixing that now

Reply:
OK. Maybe add a comment specifying why we don't want to consider m and y units as valid for our application.

Reply:
Sure - adding